tests/unit/aofrw.tcl

```diff
@@ -179,24 +179,41 @@ start_server {tags {"aofrw external:skip"} overrides {aof-use-rdb-preamble no}}
     }
 
     test {BGREWRITEAOF is delayed if BGSAVE is in progress} {
-        r multi
+        set master [srv 0 client]
+        set master_host [srv 0 host]
+        set master_port [srv 0 port]
+        # Write big data to delay bgsave and bgrewrite
+        set load_handle0 [start_write_load $master_host $master_port 10]
+        set load_handle1 [start_write_load $master_host $master_port 10]
+        set load_handle2 [start_write_load $master_host $master_port 10]
+        set load_handle3 [start_write_load $master_host $master_port 10]
+        set load_handle4 [start_write_load $master_host $master_port 10]
+        wait_for_condition 1000 100 {
+            [r dbsize] > 100000
+        } else {
+            fail "No write load detected."
+        }
```

Comment:
you don't need so much data.
enough to add one key and use `rdb-key-save-delay` config (set to 10000000).
then after observing it got scheduled, you revert that config and kill the bgsave child (`exec kill -9 [get_child_pid -1]`)

Reply:
I know, but we still have two `bgrewriteaof` tests, we currently don’t have `aof-rewrite-delay`

Reply:
i think we can use `rdb_key_save_delay` in aof.c
it's badly named, i should have dropped the `rdb` part when i created it, but it's just a debug config for testing...

also, i don't seem to understand that second test.
when BGREWRITEAOF is called while there's an AOF child, it doesn't turn on the `scheduled` flag, just responds with an error. what am i missing about this test?

Reply:
The logic you describe only  true if the child process type is CHILD_TYPE_RDB.
```
if (server.child_type == CHILD_TYPE_AOF) {
        addReplyError(c,"Background append only file rewriting already in progress");
    } else if (hasActiveChildProcess()) {
        server.aof_rewrite_scheduled = 1;
        addReplyStatus(c,"Background append only file rewriting scheduled");
    } else if (rewriteAppendOnlyFileBackground() == C_OK) {
```

Reply:
@oranagra Maybe add a `aof-key-save-delay` config is better choice ？

Reply:
@chenyang8094 i don't think we want another config. just ignore the name and use it in AOFRW.
or we can add an alias for that config (i.e. rename it to a plain `key-save-delay`) if it makes us feel better (we don't need two configs)

alternatively (or additionally) you can just re-enable `aof-use-rdb-preamble` which was left set to `no` in the last test (unintentionally).

i still don't understand what you said about the test.. we only set the scheduled flag if the child is **not** AOF, and the test actually checks that he got the "already in progress" error. so i don't understand why the scheduled flag is set and tested.

Reply:
Oh，I think `aof-use-rdb-preamble `can solve the problem.